ledger/sandbox/src/main/scala/com/digitalasset/platform/indexer/StandaloneIndexerServer.scala

```diff
@@ -21,44 +21,40 @@ final class StandaloneIndexerServer(
 
   private val logger = ContextualizedLogger.get(this.getClass)
 
-  override def acquire()(implicit executionContext: ExecutionContext): Resource[Unit] =
-    for {
-      // ActorSystem name not allowed to contain daml-lf LedgerString characters ".:#/ "
-      actorSystem <- AkkaResourceOwner
-        .forActorSystem(() =>
-          ActorSystem("StandaloneIndexerServer-" + config.participantId.filterNot(".:#/ ".toSet)))
-        .acquire()
-      indexerFactory = new JdbcIndexerFactory(config.jdbcUrl, metrics)
-      indexer = new RecoveringIndexer(
-        actorSystem.scheduler,
-        config.restartDelay,
-      )
-      _ <- config.startupMode match {
-        case IndexerStartupMode.MigrateOnly =>
-          Resource.successful(Future.unit)
-        case IndexerStartupMode.MigrateAndStart =>
-          Resource
-            .fromFuture(indexerFactory.migrateSchema(config.allowExistingSchema))
-            .flatMap(startIndexer(indexer, _, actorSystem))
-        case IndexerStartupMode.ValidateAndStart =>
-          Resource
-            .fromFuture(indexerFactory.validateSchema())
-            .flatMap(startIndexer(indexer, _, actorSystem))
-      }
-    } yield {
-      logger.debug("Waiting for indexer to initialize the database")
+  override def acquire()(implicit executionContext: ExecutionContext): Resource[Unit] = {
+    val indexerFactory = new JdbcIndexerFactory(config.jdbcUrl, metrics)
+    val indexer = new RecoveringIndexer(actorSystem.scheduler, config.restartDelay)
+    config.startupMode match {
+      case IndexerStartupMode.MigrateOnly =>
+        Resource.successful(())
+      case IndexerStartupMode.MigrateAndStart =>
+        Resource
+          .fromFuture(indexerFactory.migrateSchema(config.allowExistingSchema))
+          .flatMap(startIndexer(indexer, _, actorSystem))
+          .map { _ =>
+            logger.debug("Waiting for the indexer to initialize the database.")
+          }
+      case IndexerStartupMode.ValidateAndStart =>
+        Resource
+          .fromFuture(indexerFactory.validateSchema())
+          .flatMap(startIndexer(indexer, _, actorSystem))
+          .map { _ =>
+            logger.debug("Waiting for the indexer to initialize the database.")
```

Comment:
Since the future of the indexer startup is discarded in line 59, where does the "waiting" actually happen now?

Reply:
That future resolves when the indexer is _complete_. [We always discarded it](https://github.com/digital-asset/daml/pull/4827/files#diff-9d56581c46fe29b6c225c0425dc4f49eL36); it's really only there for testing.

Reply:
Right, I see that now. Thanks.